SPEC.md

```diff
@@ -0,0 +1,195 @@
+
+```
+{
+    "cdiVersion": "0.1.0",
+    "kind": "<name>",
+    "container-runtime": ["<container-runtime-name>"], (optional)
+
+    "cdiDevices": [
+        {
+            "name": "<name>",
+            "nameShort": ["<short-name>", "<short-name>"], (optional)
+
+            // Same as the below containerSpec field.
+            // This field should only be applied to the Container's OCI spec
+            // if that specific device is requested.
+            "containerSpec": { ... }
+        }
+    ],
+
+    // This field should be applied to the Container's OCI spec if any of the
+    // devices defined above are requested on the CLI
+    "containerSpec": [
+        {
+            "devices": [ (optional)
+                {
+                    "hostPath": "<path>",
+                    "containerPath": "<path>",
+
+                    // Cgroups permissions of the device, candidates are one or more of
+                    // * r - allows container to read from the specified device.
+                    // * w - allows container to write to the specified device.
+                    // * m - allows container to create device files that do not yet exist.
+                    "permissions": "<permissions>" (optional)
+                }
+            ]
+            "mounts": [ (optional)
+                {
+                    "hostPath": "<source>",
+                    "containerPath": "<destination>",
+                    "options": "<OCI Mount Options>", (optional)
+                }
+            ],
+            "hooks": [ (optional)
+                {
+                    "hookName": "<hookName>",
+                    "path": "<path>",
+                    "args": ["<arg>", "<arg>"], (optional)
+                    "env":  [ "<envName>=<envValue>"], (optional)
+                    "timeout": <int> (optional)
+                }
+            ]
+        }
+    ]
+}
+```
+
+#### Specification version
+
+* `cdiVersion` (string, REQUIRED) MUST be in [Semantic Version 2.0](https://semver.org) and specifies the version of the CDI specification used by the vendor.
+
+#### Kind
+
+* `kind` (string, REQUIRED) field specifies a label which uniquely identifies the device vendor.
+  It can be used to disambiguate the vendor that matches a device, e.g: `docker/podman run --device vendor.com/device=foo ...`.
+    * The `kind` and `kindShort` labels have two segments: a prefix and a name, separated by a slash (/).
+    * The name segment is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (\_), dots (.), and alphanumerics between.
+    * The prefix must be a DNS subdomain: a series of DNS labels separated by dots (.), not longer than 253 characters in total, followed by a slash (/).
+    * Examples (not an exhaustive list):
+      * Valid: `vendor.com/foo`, `foo.bar.baz/foo-bar123.B_az`.
+      * Invalid: `foo`, `vendor.com/foo/`, `vendor.com/foo/bar`.
+
```

Comment:
Does it make sense to have a sanity check for the hooks, if device  A.B/C is requested that only hooks that are actually from A.B/C are executed?

Reply:
I'm not sure I got your question for this one, if device "vendor.com/foo" is requested, then the runtime reads the json that specified `kind: "vendor.com/foo"` and merges in the OCI spec the containerSpec that matches that device as well as the containerSpec field for this vendor.

Reply:
I was just thinking of users copy and paste specs and forgetting to change the hook binaries and the runtime wants to activate a GPU with the hooks of an FPGA.

The runtime could spit out, you're using the wrong hooks for this device.

Reply:
I'm not sure how we could detect that. Are you suggesting that hooks executable have a specific name?